Scripts/script-url-parse.yml

```diff
@@ -0,0 +1,45 @@
+commonfields:
+  id: 6e22a899-44c3-4b7f-8bf7-878439ef2c93
+  version: 32
+name: cs-url-parse
+script: |-
+  import re
+
+  def agent_extract(url):
+      '''
+      Use regex to extract the CrowdStrike Agent ID (aid) from a CrowdStrike alert URL.
+      The output of this can be passed to cs-device-details for a host lookup.
+
+      url = 'https://falcon.crowdstrike.com/activity/detections/detail/9de027b146528z357a1a01e9cdf55724/176094545445?_cid=41ffdgime3514d6fa68df613d7b1937d'
+
+      # Find the Agent ID and map it to the "aid" group name.
+      match = re.search(r'detail\/(?P<aid>(\w{32}))\/', url)
+
+      match.group('aid')
+      # returns '9de027b146528z357a1a01e9cdf55724'
+      '''
+
+      match = re.search(r'detail\/(?P<aid>(\w{32}))\/', url)
+      aid = match.group('aid')
+      return aid
+
+  agent_id = agent_extract(demisto.args()['url'])
+  demisto.setContext('agent_id', agent_id)
+  demisto.results(agent_id)
```

Comment:
do 

```
demisto.results({
    'Type': entryTypes['note'],
    'Contents': agent_id,
    'ContentsFormat': formats['json'],
    'ReadableContentsFormat': formats['text'],
    'HumanReadable': agent_id if agent_id else '{} don\'t contains agent id'
})
```
in case the url will not contain agent id, we should print some human readable text to user